README.md
# icon_forest

- A collection of 160k+ icons from 230+ icon libraries.
- It makes icons from all pub.dev packages searchable in one place.
- You can search, select and apply the icons to your Flutter projects.

<img alt="sample icons screenshot" src="https://raw.githubusercontent.com/iconforest/icon_forest/main/readme_images/icons_snapshot.png" />

It has two components:
1. a free app that searches icons (**IconForest** app)
2. a collection of icon libraries (**icon_forest** package)

---

# IconForest app <img alt="apk" src="https://raw.githubusercontent.com/iconforest/icon_forest/main/readme_images/logo.png"/>

<img alt="search icons, mark favorites, output icon code" src="https://raw.githubusercontent.com/iconforest/icon_forest/main/readme_images/demo.gif" />

Icons can be easily applied to your projects with pubspec.yaml.

![icons code viewer pasteboard example](https://raw.githubusercontent.com/iconforest/icon_forest/main/readme_images/codeviewer_snapshot.png)

<table>
  <tbody>
    <tr>
      <td align="center">
        <a
          href="https://iconforest.flutterplay.com/download/googleplay.php"
          ><img
            alt="Google Play"
            src="https://raw.githubusercontent.com/iconforest/icon_forest/main/readme_images/download/googleplay.png"
        /></a>
      </td>
      <td align="center">
        <a href="https://iconforest.flutterplay.com/download/android.php"
          ><img
            alt="apk"
            src="https://raw.githubusercontent.com/iconforest/icon_forest/main/readme_images/download/apk.png"
        /></a>
      </td>
      <td align="center">
        <a href="https://iconforest.flutterplay.com/download/appstore.php"
          ><img
            alt="Appstore"
            src="https://raw.githubusercontent.com/iconforest/icon_forest/main/readme_images/download/appstore.png"
        /></a>
      </td>
    </tr>
    <tr>
      <td align="center">
        <img
          alt="Google Play"
          src="https://raw.githubusercontent.com/iconforest/icon_forest/main/readme_images/qr/qr_googleplay.png"
          width="127"
          height="127"
        />
      </td>
      <td align="center">
        <img
          alt="apk"
          src="https://raw.githubusercontent.com/iconforest/icon_forest/main/readme_images/qr/qr_apk.png"
          width="127"
          height="127"
        />
      </td>
      <td align="center">
        <img
          alt="Appstore"
          src="https://raw.githubusercontent.com/iconforest/icon_forest/main/readme_images/qr/qr_appstore.png"
          width="128"
          height="128"
        />
      </td>
    </tr>
  </tbody>
</table>

---

# Icon libraries

It contains 41 icon libraries (38k+ icons). They come from:
- converting existing non-null-safety or invalid icon packages
- open-source icon libraries which are new to pub.dev


You can search the following icon libraries with IconForest app:

||||||
| :- | :- |:- | :- | :- |
 |Academicons (149 icons)|Adinkra Symbols (99 icons)|Adwaita Icon (956 icons)|Akar Icons (381 icons)|Ali Icons (71 icons)|
 |AmazingNeoIcons (112 icons)|Ant Design (1829 icons)|Arcticons (4237 icons)|Bootstrap Icons (1682 icons)|Boxicons (1525 icons)|
 |Mbi linecons (48 icons)|BoxIcons Logo (153 icons)|bpmn (106 icons)|Brand Logos (924 icons)|Brandico (45 icons)|
 |Bytesize (101 icons)|Carbon design (1584 icons)|Charm Icons (250 icons)|Clarity Design (1121 icons)|codicon (423 icons)|
 |CoolIcons (412 icons)|CoreUI Brands (831 icons)|CoreUI Free (562 icons)|Crypto Icons (color) (30 icons)|Crypto Icons (font) (282 icons)|
 |Crypto Icons (font) (477 icons)|css.gg (704 icons)|Cupertino Icons  (1322 icons)|Dashicons (347 icons)|Devicon (369 icons)|
 |Elegant (100 icons)|Element Plus (283 icons)|Elusive Icons (304 icons)|Emoji (20 icons)|Emoji One (2159 icons)|
 |EmojioneMonotone (1713 icons)|EmojioneV1 (1529 icons)|Emojis (46 icons)|Entypo (411 icons)|Entypo+ Social (76 icons)|
 |EOS Icons (251 icons)|Eva Icons (490 icons)|Evil Icons (70 icons)|Feather Icon (286 icons)|File Icons (930 icons)|
 |Flag (192 icons)|Flat Icons (Arrows Set) (159 icons)|Flat Icons (Medium) (380 icons)|Flat Icons (Social) (40 icons)|Flat UI (98 icons)|
 |FluentUI system icons (5715 icons)|Font Awesome 4 (786 icons)|Font Awesome 5 (brand) (924 icons)|Font Awesome 5 (regular) (162 icons)|Font Awesome 5 (solid) (138 icons)|
 |Font-GIS (344 icons)|FontAudio (155 icons)|Fontelico (34 icons)|Fontisto (618 icons)|Foundation Icon Fonts 3 (283 icons)|
 |Framework7 Icons (1228 icons)|Frino (384 icons)|Fxemoji (1034 icons)|Gala Icons (51 icons)|Geira Icons (197 icons)|
 |GeoGlyphs (30 icons)|Gridicons (201 icons)|Grommet Icons (615 icons)|Health Icons (1754 icons)|Helium (112 icons)|
 |Hero (Outline) (230 icons)|Hero (Solid) (230 icons)|Humanitarian icons (307 icons)|Icofont (2095 icons)|IcoMoon (491 icons)|
 |Icon Park (2437 icons)|IconaMoon (304 icons)|Iconic (223 icons)|Iconly (Bold) (100 icons)|Iconly (Broken) (100 icons)|
 |Iconly (Light) (100 icons)|iconoir (1091 icons)|IconPark Outline (2437 icons)|icons8 (234 icons)|Icons8 Windows 8 Icons (234 icons)|
 |iconsax (2648 icons)|Innowatio font icons (105 icons)|IonIcons (3681 icons)|Jam Icons (940 icons)|Ligature Symbols (348 icons)|
 |Line Awesome Icons (1544 icons)|Linear Icons Free (175 icons)|Logos (1583 icons)|Lucide (624 icons)|Majesticons (760 icons)|
 |Maki (595 icons)|Map Icons (16 icons)|Material Design Icon (Community) (6595 icons)|Material Design Iconic Font (885 icons)|Mbi handmade (96 icons)|
 |Mbi file (50 icons)|Material Line Icons (235 icons)|Medical Icons (144 icons)|Meteocons (47 icons)|MFG Labs (153 icons)|
 |Modern Pictograms (91 icons)|Mono Icons (180 icons)|Navigation Maps (16 icons)|Nimbus (134 icons)|Noto Emoji (3798 icons)|
 |Noto Emoji (v1) (2319 icons)|OctIcons (692 icons)|OOUI (322 icons)|Open Iconic (223 icons)|OpenMoji (4045 icons)|
 |Outline Material Icons (1010 icons)|Pepicons (316 icons)|Phosphor (6312 icons)|Pixelarticons (460 icons)|PrestaShop (479 icons)|
 |primeicons (238 icons)|Quill Icons (140 icons)|Radix Icons (331 icons)|Raphael (266 icons)|Remix (2271 icons)|
 |Simple Icons (2379 icons)|Simple Line Icon (194 icons)|SmartIcons Glyph (803 icons)|Subway Icon Set (308 icons)|System UIcons (420 icons)|
 |Tabler icons (1298 icons)|Teenyicons (1195 icons)|Ternav Icon (Bold) (184 icons)|Ternav Icon (Light outline) (184 icons)|Ternav Icon (Light) (184 icons)|
 |Themify Icons (352 icons)|TopCoat Icons (89 icons)|Twemoji (4406 icons)|Twicon (100 icons)|Typicons (336 icons)|
 |uiw icons (214 icons)|Unicons (Line) (1206 icons)|Unicons (Solid) (189 icons)|Unicons (Thinline) (214 icons)|Unicons Monochrome (296 icons)|
 |Vaadin Icons (636 icons)|Vesper Icons (172 icons)|VSCode Icons (1110 icons)|Mbi emoji (50 icons)|Weather Icons (601 icons)|
 |Weather Icons (590 icons)|Web/Symbol (1228 icons)|WebHostingHub Glyphs (2125 icons)|Yaru (415 icons)|Zocial (100 icons)|
 |Zondicons (301 icons)|basf (80 icons)|Mbi combi (129 icons)|||


---

## Our goal

Make beautiful designs faster for Flutter developers. 
To achieve this, we:
- collect icon packages from pub.dev and make them searchable in one place
- test all icon packages, correct sample codes, and remove invalid icons
- make duplicated icon libraries invisible by default
- explore new beautiful icon libraries
- update the icon libraries to provide you with the richest and latest icons collections

## Contact us
<a href="mailto:iconforest@flutterplay.com">iconforest@flutterplay.com</a>



CHANGELOG.md
## 1.1.7

- Fine tune documentation

## 1.1.2

- Add 5 icon libraries

## 1.1

- Make 6 icon libraries searchable

## 1.0.3

- Fix the path of svg libraries

## 1.0.2

- Fix svg from bytesize, charm_icons, iconoir, system_uicons libraries

## 1.0.1

- Fix display of several libraries

## 1.0.0+1

- Initial release



lib/bytesize.dart
import 'package:flutter/material.dart';
import 'package:flutter_svg/flutter_svg.dart';

import 'icon_forest.dart';

/// Bytesize icons (101 icons)
///
/// - build svg for bytesize icon libraries
/// App crypto icons (30 icons)
///
/// - build svg for app_crypto_icons icon libraries
/// - constants link to svg icon resources
class Bytesize extends StatelessWidget {
  final String assetName;
  final bool matchTextDirection;
  final AssetBundle? bundle;
  final String? package;
  final double? width;
  final double? height;
  final BoxFit fit;
  final AlignmentGeometry alignment;
  final bool allowDrawingOutsideViewBox;
  final WidgetBuilder? placeholderBuilder;
  final Color? color;
  final BlendMode colorBlendMode;
  final String? semanticsLabel;
  final bool excludeFromSemantics;
  final Clip clipBehavior;
  final bool cacheColorFilter;
  final SvgTheme? theme;

  const Bytesize(this.assetName,
      {Key? key,
      this.matchTextDirection = false,
      this.bundle,
      this.package,
      this.width,
      this.height,
      this.fit = BoxFit.contain,
      this.alignment = Alignment.center,
      this.allowDrawingOutsideViewBox = false,
      this.placeholderBuilder,
      this.color,
      this.colorBlendMode = BlendMode.srcIn,
      this.semanticsLabel,
      this.excludeFromSemantics = false,
      this.clipBehavior = Clip.hardEdge,
      this.cacheColorFilter = false,
      this.theme})
      : super(key: key);

  @override
  Widget build(BuildContext context) {
    return IconForest.svgAsset(
      'bytesize',
      assetName,
      key: key,
      matchTextDirection: matchTextDirection,
      bundle: bundle,
      package: package,
      width: width,
      height: height,
      fit: fit,
      alignment: alignment,
      allowDrawingOutsideViewBox: allowDrawingOutsideViewBox,
      placeholderBuilder: placeholderBuilder,
      color: color,
      colorBlendMode: colorBlendMode,
      semanticsLabel: semanticsLabel,
      excludeFromSemantics: excludeFromSemantics,
      clipBehavior: clipBehavior,
      cacheColorFilter: cacheColorFilter,
      theme: theme,
    );
  }

  /// map activity to code point activity
  static const String activity = 'activity';

  /// map alert to code point alert
  static const String alert = 'alert';

  /// map archive to code point archive
  static const String archive = 'archive';

  /// map arrow_bottom to code point arrow_bottom
  static const String arrow_bottom = 'arrow_bottom';

  /// map arrow_left to code point arrow_left
  static const String arrow_left = 'arrow_left';

  /// map arrow_right to code point arrow_right
  static const String arrow_right = 'arrow_right';

  /// map arrow_top to code point arrow_top
  static const String arrow_top = 'arrow_top';

  /// map backwards to code point backwards
  static const String backwards = 'backwards';

  /// map bag to code point bag
  static const String bag = 'bag';

  /// map ban to code point ban
  static const String ban = 'ban';

  /// map bell to code point bell
  static const String bell = 'bell';

  /// map book to code point book
  static const String book = 'book';

  /// map bookmark to code point bookmark
  static const String bookmark = 'bookmark';

  /// map calendar to code point calendar
  static const String calendar = 'calendar';

  /// map camera to code point camera
  static const String camera = 'camera';

  /// map caret_bottom to code point caret_bottom
  static const String caret_bottom = 'caret_bottom';

  /// map caret_left to code point caret_left
  static const String caret_left = 'caret_left';

  /// map caret_right to code point caret_right
  static const String caret_right = 'caret_right';

  /// map caret_top to code point caret_top
  static const String caret_top = 'caret_top';

  /// map cart to code point cart
  static const String cart = 'cart';

  /// map checkmark to code point checkmark
  static const String checkmark = 'checkmark';

  /// map chevron_bottom to code point chevron_bottom
  static const String chevron_bottom = 'chevron_bottom';

  /// map chevron_left to code point chevron_left
  static const String chevron_left = 'chevron_left';

  /// map chevron_right to code point chevron_right
  static const String chevron_right = 'chevron_right';

  /// map chevron_top to code point chevron_top
  static const String chevron_top = 'chevron_top';

  /// map clipboard to code point clipboard
  static const String clipboard = 'clipboard';

  /// map clock to code point clock
  static const String clock = 'clock';

  /// map close to code point close
  static const String close = 'close';

  /// map code to code point code
  static const String code = 'code';

  /// map compose to code point compose
  static const String compose = 'compose';

  /// map creditcard to code point creditcard
  static const String creditcard = 'creditcard';

  /// map desktop to code point desktop
  static const String desktop = 'desktop';

  /// map download to code point download
  static const String download = 'download';

  /// map edit to code point edit
  static const String edit = 'edit';

  /// map eject to code point eject
  static const String eject = 'eject';

  /// map ellipsis_horizontal to code point ellipsis_horizontal
  static const String ellipsis_horizontal = 'ellipsis_horizontal';

  /// map ellipsis_vertical to code point ellipsis_vertical
  static const String ellipsis_vertical = 'ellipsis_vertical';

  /// map end to code point end
  static const String end = 'end';

  /// map export_ to code point export
  static const String export_ = 'export';

  /// map external_ to code point external
  static const String external_ = 'external';

  /// map eye to code point eye
  static const String eye = 'eye';

  /// map feed to code point feed
  static const String feed = 'feed';

  /// map file to code point file
  static const String file = 'file';

  /// map filter to code point filter
  static const String filter = 'filter';

  /// map flag to code point flag
  static const String flag = 'flag';

  /// map folder_open to code point folder_open
  static const String folder_open = 'folder_open';

  /// map folder to code point folder
  static const String folder = 'folder';

  /// map forwards to code point forwards
  static const String forwards = 'forwards';

  /// map fullscreen_exit to code point fullscreen_exit
  static const String fullscreen_exit = 'fullscreen_exit';

  /// map fullscreen to code point fullscreen
  static const String fullscreen = 'fullscreen';

  /// map gift to code point gift
  static const String gift = 'gift';

  /// map github to code point github
  static const String github = 'github';

  /// map heart to code point heart
  static const String heart = 'heart';

  /// map home to code point home
  static const String home = 'home';

  /// map import_ to code point import
  static const String import_ = 'import';

  /// map inbox to code point inbox
  static const String inbox = 'inbox';

  /// map info to code point info
  static const String info = 'info';

  /// map lightning to code point lightning
  static const String lightning = 'lightning';

  /// map link to code point link
  static const String link = 'link';

  /// map location to code point location
  static const String location = 'location';

  /// map lock to code point lock
  static const String lock = 'lock';

  /// map mail to code point mail
  static const String mail = 'mail';

  /// map menu to code point menu
  static const String menu = 'menu';

  /// map message to code point message
  static const String message = 'message';

  /// map microphone to code point microphone
  static const String microphone = 'microphone';

  /// map minus to code point minus
  static const String minus = 'minus';

  /// map mobile to code point mobile
  static const String mobile = 'mobile';

  /// map moon to code point moon
  static const String moon = 'moon';

  /// map move to code point move
  static const String move = 'move';

  /// map music to code point music
  static const String music = 'music';

  /// map mute to code point mute
  static const String mute = 'mute';

  /// map options to code point options
  static const String options = 'options';

  /// map paperclip to code point paperclip
  static const String paperclip = 'paperclip';

  /// map pause to code point pause
  static const String pause = 'pause';

  /// map photo to code point photo
  static const String photo = 'photo';

  /// map play to code point play
  static const String play = 'play';

  /// map plus to code point plus
  static const String plus = 'plus';

  /// map portfolio to code point portfolio
  static const String portfolio = 'portfolio';

  /// map print to code point print
  static const String print = 'print';

  /// map reload to code point reload
  static const String reload = 'reload';

  /// map reply to code point reply
  static const String reply = 'reply';

  /// map search to code point search
  static const String search = 'search';

  /// map send to code point send
  static const String send = 'send';

  /// map settings to code point settings
  static const String settings = 'settings';

  /// map sign_in to code point sign_in
  static const String sign_in = 'sign_in';

  /// map sign_out to code point sign_out
  static const String sign_out = 'sign_out';

  /// map star to code point star
  static const String star = 'star';

  /// map start to code point start
  static const String start = 'start';

  /// map tag to code point tag
  static const String tag = 'tag';

  /// map telephone to code point telephone
  static const String telephone = 'telephone';

  /// map trash to code point trash
  static const String trash = 'trash';

  /// map twitter to code point twitter
  static const String twitter = 'twitter';

  /// map unlock to code point unlock
  static const String unlock = 'unlock';

  /// map upload to code point upload
  static const String upload = 'upload';

  /// map user to code point user
  static const String user = 'user';

  /// map video to code point video
  static const String video = 'video';

  /// map volume to code point volume
  static const String volume = 'volume';

  /// map work to code point work
  static const String work = 'work';

  /// map zoom_in to code point zoom_in
  static const String zoom_in = 'zoom_in';

  /// map zoom_out to code point zoom_out
  static const String zoom_out = 'zoom_out';

  /// map zoom_reset to code point zoom_reset
  static const String zoom_reset = 'zoom_reset';
}
